frontend/public/components/app.jsx

```diff
@@ -141,7 +143,10 @@ const App = (props) => {
 
   const onCSPViolation = React.useCallback((event) => {
     // eslint-disable-next-line no-console
-    console.warn('Content Security Policy violation detected', event);
+    console.warn(CSP_VIOLATION_MESSAGE, event);
+
+    // Store CSP violation message in global variable for e2e test purposes
+    window.windowCSPWarning = CSP_VIOLATION_MESSAGE;
```

Comment:
you may want to update the window type: https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/%40types/sdk/index.d.ts